### PR TITLE
user settings i18n

### PR DESCRIPTION
**Pull request by yheuhtozr**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Extracts strings for i18n from setting pages in app/views/users. Attached fr locales for testing purposes. Existing translations up to #15055 reflected (hopefully).

⚠ As a side effect, #14905 is reverted. This is because the commit tried to localize the constants in English words, and broke setting tabs UI in the French locale, which always expects English named variables. This PR only compensates for its tab label L10N, but not localized path URL (because it needs a route i18n mechanism and does not work by itself either way).

## Related Tickets & Documents

#14888, #14905

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [X] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/users_controller.rb
Comment (yheuhtozr) [reply app/controllers/users_controller.rb:270]: In the controller i18n phrase we will frequently see English strings defined as constants in the code base. I usually only directly i18n them if they are simply used as UI strings. When they work as any parameters, I instead keep them as such intact and replace via dynamic i18n keys or such.

**Review by citizen428 — APPROVED** [review]

**Comment by yheuhtozr** [comment]
@citizen428 Is this ready to have the second reviewer?

**Comment by msarit** [comment]
@yheuhtozr Hello! 👋🏾  Thank you for all your hard work on the I18n project!

Please fix the conflicts; we'll work on getting you a second review 🙏🏾

**Review by jeremyf — APPROVED** [review]

**Comment by jeremyf** [comment]
I'm hesitant to merge this today (and frankly any time this week due to holidays in the USA).  But have this to review for Monday and lay out a plan.

Perhaps @citizen428 is ready to pull the trigger?

**Comment by jeremyf** [comment]
Sorry about not getting back to this.  Consider my hesitation "rescinded"

**Pull request merged**
